tests/test_wizard_postgres.py

```diff
@@ -87,3 +95,84 @@ def test_postgres():
     assert 'title' in str(ref_list)
     assert 'url' in str(ref_list)
     assert 'research-hub.nrel.gov' in str(ref_list)
+
+
+def test_ref_replace(mocker):
+    """Test to ensure removal of double quotes from references."""
+    mock_conn_cm = mocker.MagicMock()
+    mock_conn = mock_conn_cm.__enter__.return_value
+    mock_conn.cursor.return_value = Cursor()
+
+    mock_connect = mocker.patch('psycopg2.connect')
+    mock_connect.return_value = mock_conn_cm
+
+    wizard = EnergyWizardPostgres(db_host='Dummy', db_port='Dummy',
+                                  db_name='Dummy', db_schema='Dummy',
+                                  db_table='Dummy',
+                                  boto_client=BotoClient(),
+                                  meta_columns=['title', 'url', 'id'])
+
+    refs = [(chr(34), 'test.com', '5a'),
+            ('remove "double" quotes', 'test_2.com', '7b')]
+
+    ids = np.array(['7b', '5a'])
+
+    out = wizard._format_refs(refs, ids)
+
+    assert len(out) > 1
+    for i in out:
+        assert json.loads(i)
+
+
+def test_ids(mocker):
+    """Test to ensure only records with valid ids are returned."""
+    mock_conn_cm = mocker.MagicMock()
+    mock_conn = mock_conn_cm.__enter__.return_value
+    mock_conn.cursor.return_value = Cursor()
+
+    mock_connect = mocker.patch('psycopg2.connect')
+    mock_connect.return_value = mock_conn_cm
+
+    wizard = EnergyWizardPostgres(db_host='Dummy', db_port='Dummy',
+                                  db_name='Dummy', db_schema='Dummy',
+                                  db_table='Dummy',
+                                  boto_client=BotoClient(),
+                                  meta_columns=['title', 'url', 'id'])
+
+    refs = [('title', 'test.com', '5a'),
+            ('title2', 'test_2.com', '7b')]
+
+    ids = np.array(['7c', '5a'])
+
+    out = wizard._format_refs(refs, ids)
+
+    assert len(out) == 1
+    assert '7b' not in out
```

Comment:
I think you probably meant for this to be 
```suggestion
    assert '7b' not in out[0]
```

Otherwise, if `out = ["... 7b ..."]`, then `'7b' not in out` still evaluates to `True`

Reply:
In this case I am checking to ensure an id doesn't appear in the reference list if it wasn't in the list of ids. So I think I want to test the complete output not just the first item. For example, '7b' isn't in the ids list so it shouldn't be included in the output at all. Does that make sense?

Reply:
Yes, but in the line right above it, you `assert len(out) == 1`, which means `out[0]` _is_ the entire output.

But let's say `out` did have more than one item. In that case, `assert '7b' not in out` does not work! Try it with this example:

```
out = ["My ID is 7b", "My ID is also 7b", '{"id": "7b"}']
assert '7b' not in out  # this passes, even though '7b' appears in every item
```

What you are probably looking for then is something like `assert not any('7b' in item for item in out)`

Reply:
P.S. the only time your assert would behave as you are expecting is if `out` literally had the string `7b` in it. e.g. `out = ["a", "b", "7b"]`